spec/ics-007-tendermint-client/README.md

```diff
@@ -140,17 +158,17 @@ function checkMisbehaviourAndUpdateState(
     assert(evidence.h1.height === evidence.h2.height)
     // assert that the commitments are different
     assert(evidence.h1.commitmentRoot !== evidence.h2.commitmentRoot)
-    // fetch the previously verified commitment root & validator set hash
+    // fetch the previously verified commitment root & validator set
     consensusState = get("clients/{identifier}/consensusStates/{evidence.fromHeight}")
-    // check that the validator set matches
-    assert(consensusState.validatorSetHash === evidence.fromValidatorSet.hash())
+    // assert that the timestamp is not from more than an unbonding period ago
+    assert(currentTimestamp() - consensusState.timestamp < clientState.unbondingPeriod)
     // check if the light client "would have been fooled"
     assert(
-      verify(evidence.fromValidatorSet, evidence.fromHeight, evidence.h1) &&
-      verify(evidence.fromValidatorSet, evidence.fromHeight, evidence.h2)
+      verify(consensusState.validatorSet, evidence.fromHeight, evidence.h1) &&
+      verify(consensusState.validatorSet, evidence.fromHeight, evidence.h2)
       )
     // set the frozen height
-    clientState.frozenHeight = evidence.h1.height // which is same as h2.height
+    clientState.frozenHeight = min(clientState.frozenHeight, evidence.h1.height) // which is same as h2.height
```

Comment:
shouldn't this be `max`? `clientState` always starts with `frozenHeight = 0`

Reply:
No, it should be `min`, since we want to freeze at an earlier height if misbehaviour is discovered. We should check if it is `0` though first, you're right.

Reply:
see? the `IsFrozen()` function is handy 😜

Reply:
indeed!

Reply:
I updated the implementation to take this into account